ejb3/src/main/java/org/jboss/as/ejb3/iiop/EjbCorbaServant.java

```diff
@@ -314,21 +324,59 @@ public OutputStream _invoke(final String opName, final InputStream in, final Res
                                 retVal = false;
                             }
                         } else {
-
-                            if (sc != null) {
-                                setSecurityContextOnAssociation(sc);
+                            if (this.securityDomain != null) {
+                                // an elytron security domain is available: authenticate and authorize the client before
+                                // invoking the component.
+                                ServerAuthenticationContext context = this.securityDomain.createNewAuthenticationContext();
+                                if (credential == sasCurrent) {
+                                    // TODO change this to use the SASCurrent as the credential when we have a realm that can act on it.
```

Comment:
If you get a principal with no credentials, then we could skip the evidence verify (if that's appropriate i.e. the source of this thing is trusted (this is basically the same as using the name as the password anyway)), otherwise we can reject the authentication.  Otherwise if there's some specific other evidence type we support that we should be testing for, we can add that.